src/operator/contrib/cu_open_lstm_rnn-inl.cuh

```diff
@@ -0,0 +1,1125 @@
+                      1.0, 1.0);
+          matmul_stridedbatched_ex(m_cublas_handle,
+                                   RE_CAST(m_data_T_major_grad.dptr) +
+                                     (seq_rend + 1) * param_.embed_dim * param_.batch_size,
+                                   i2h_weight.dptr_,
+                                   RE_CAST(m_i2h_workspace.dptr) +
+                                     (seq_rend + 1) * 4 * m_num_hidden_units_x_batch_size,
+                                   CUBLAS_OP_T, CUBLAS_OP_N,
+                                   4 * param_.num_hidden_units, param_.embed_dim,
+                                   4 * param_.num_hidden_units, param_.batch_size,
+                                   param_.embed_dim * param_.batch_size,
+                                   0,
+                                   4 * m_num_hidden_units_x_batch_size,
+                                   seq_rbegin - seq_rend,
+                                   1.0, 0.0);
+        }  // if (layer_idx > 0)
+      }  // layer_idx
+    }  // schedule
+    CUDA_CALL(cudaMemcpyAsync(i2h_bias_grad.dptr_,
+                              h2h_bias_grad.dptr_,
+                              param_.num_layers * 4 * param_.num_hidden_units * sizeof(float),
+                              cudaMemcpyDeviceToDevice,
+                              m_stream_h2h[0]));
+    /*
+    transpose(m_cublas_handle,
+      m_data_B_major_grad, m_data_T_major_grad,
+      m_seq_len * m_embed_dim, m_batch_size);
+     */
+    transpose(m_cublas_handle,
+              data_grad.dptr_,
+              RE_CAST(m_data_T_major_grad.dptr),
+              param_.seq_len * param_.embed_dim,
+              param_.batch_size);
+    CUDA_CALL(cudaStreamSynchronize(m_stream_i2h[0]));
+    CUDA_CALL(cudaStreamSynchronize(m_stream_h2h[0]));
+  }
+
+ private:
+  void Init(mshadow::Stream<gpu> *s,
+            const std::vector<TBlob> &in_data,
+            const std::vector<TBlob> &out_data) {
+    using namespace mshadow;
+
+    std::size_t in_expected = 7, out_expected = param_.state_outputs ? 3 : 1;
+
+    CHECK_EQ(in_data.size(), in_expected);
+    CHECK_EQ(out_data.size(), out_expected);
+
+    if (!initialized_) {
+      initialized_ = true;
+      Tensor<gpu, 3, float> data = in_data[open_lstm_rnn_enum::kData].get<gpu, 3, float>(s);
+      //=======================================================================
+      // infer the parameters from the input data
+      param_.batch_size = data.shape_[0];
+      param_.seq_len    = data.shape_[1];
+      param_.embed_dim  = data.shape_[2];
+      m_num_hidden_units_x_batch_size = param_.batch_size * param_.num_hidden_units;
+      //=======================================================================
+      // allocate workspace
+      m_data_T_major       = Storage::Get()->Alloc(param_.seq_len * param_.embed_dim *
+                                                     param_.batch_size * sizeof(float),
```

Comment:
nit: align

Reply:
@apeforest I deliterately indented those lines. The reason is because those are continuations of current parameter, and the following parameters are all perfectly aligned. This can help make the code more readable.